[Win Lynn] iP

weenleen opened the pull request:
### Duke 👾 

> "The way to get started is to quit talking and begin doing" - Walt Disney

Feeling overwhelmed by all your responsibilities? In constant fear of forgetting what's next on your schedule? Fear not, because Duke is here to help!

Duke is:
- easy to use
- super **fast**
- _NEVER FORGETFUL!_

Features:
- [x] Managing different tasks, such as _ToDos_, _Deadlines_ and _Events_
- [ ] Reminders (coming soon)

How do you use it? All you have to do is:
1. Download it [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html)
2. Start it up
3. Tell it what you need to remember
4. When you're done, type `bye` to say goodbye to Duke!

If you are a Java programmer, you can use it to practice Java as well! Here's the `main` method:
``` ruby
public static void main(String[] args) {
        Duke duke = new Duke("myTasks.txt");
        duke.run();
}
```

weenleen committed:
Level-1

weenleen committed:
Level-2

weenleen committed:
Level-3

weenleen committed:
Level-4

weenleen committed:
Level-4: Changed the text displayed

weenleen committed:
A-TextUiTesting

weenleen committed:
A-TextUiTesting: Part 2

weenleen committed:
Level-5

weenleen committed:
Level-6

weenleen committed:
Level-5: Added DukeException class

weenleen committed:
A-Enums: Added an enum for the types of tasks

weenleen committed:
A-Enums: updated input.txt and EXPECTED.TXT

weenleen committed:
Added Documentation

weenleen committed:
A-Enums

weenleen committed:
Level-7

weenleen committed:
Level-8

weenleen committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

weenleen committed:
Merge Level-7 and Level-8

weenleen committed:
A-MoreOOP: Create TaskList class

weenleen committed:
A-MoreOOP

weenleen committed:
A-JUnit: add test folder

weenleen committed:
A-JavaDoc

weenleen committed:
A-CodingStandard

weenleen committed:
Level-9

weenleen committed:
Merge branch 'branch-A-CodingStandard'

weenleen committed:
Merge branch 'branch-Level-9'

weenleen committed:
Add JavaDocs

weenleen committed:
Add JUnit Tests

weenleen committed:
Set up module dependencies

weenleen committed:
Add Junit tests.

aaronloh852 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,90 @@
+    /**
+     * Displays a greeting message.
+     */
+    public void greet() {
+        System.out.println("Hello from\n" + logo +"\n");
+        System.out.println("What can I do for you?");
+        showLine();
+    }
+
+    /**
+     * Displays a goodbye message.
+     */
+    public void goodBye() {
+        scanner.close();
+        System.out.println("Bye. Hope to see you again soon!");
+        showLine();
+    }
```
Maybe the method names could be phrased as an action ie. (printGreet / printGoodbye) ?

weenleen committed:
Add Gradle support

thanwinnikki commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,43 @@
+        this.taskList = new TaskList(storage.load());
+    }
+
+    public void run() {
```
Should you add javadocs to the public methods? 😄

thanwinnikki commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,66 @@
+package duke;
+
+/**
+ * A class that represents exceptions unique to duke.Duke.
```
Excellent description of the custom error class 😉

thanwinnikki commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,66 @@
+        }
+    }
+
+    /**
```
Good job with the javadocs for this class!

thanwinnikki commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,316 @@
+ * Each duke.Task has a description, and is either done or not yet done.
+ */
+public class Task {
+    private static final String DONE_STATUS_ICON = "X";
```
Excellent idea for much cleaner code.

thanwinnikki commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,316 @@
+
+                if (m_event.find()) {
+                    return new Event(m_event.group(1), m_event.group(2));
+                }else {
```
Should there be a white space before else? 🤔

thanwinnikki commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,316 @@
+        }
+
+        switch (type) {
+            case TODO:
```
Should the case indentation be 4 spaces rather than 8 for our java coding standard? 💭

thanwinnikki reviewed (commented):
Excellent code and OOP. I really like the logic of the program and it is as if you are writing code in the exact manner prof imagines it to be! I have a lot to learn from you, good job!

weenleen committed:
Add GUI

weenleen committed:
Complete basic GUI

weenleen committed:
Add greeting

weenleen committed:
Add Update Command

This allows users to update existing tasks in the TaskList, so that
they do not have to delete and re-add tasks.

weenleen committed:
Add Assertions

The Task.readTaskFromFile currently assumes that the strings in the
file are not empty or whitespaces.

This assumption may cause errors.

Using assertions will ensure that all the strings are in the valid
formats.

weenleen committed:
Merge pull request #1 from weenleen/branch-A-Assertions

Add Assertions

weenleen committed:
Task.createTaskFromInput: throw different types of exceptions.

The method currently throws the same exception for having no
description and no date /time.

This may cause users to be confused as to what is an accepted input.

Throwing other exceptions with specific error messages will inform the
user on how to use Duke properly.

weenleen committed:
Merge pull request #2 from weenleen/branch-A-CodeQuality

Task.createTaskFromInput: throw different types of exceptions.

weenleen committed:
Change file names

weenleen committed:
Update README.md

weenleen committed:
Improve GUI

weenleen committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	README.md

weenleen committed:
Add delay to exit command

weenleen committed:
Update README.md

weenleen committed:
Remove text file.

weenleen committed:
Set theme jekyll-theme-architect

weenleen committed:
Add Ui.png

weenleen committed:
Merge remote-tracking branch 'origin/master'

weenleen committed:
Set theme jekyll-theme-architect

weenleen committed:
Add more JUnit Tests.

weenleen committed:
Add Javadocs
